src/Libs/GLib-2.0/Internal/SynchronizationContext.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.Runtime.ExceptionServices;
+using System.Threading;
+
+namespace GLib.Internal;
+
+public sealed class GLibSynchronizationContext : SynchronizationContext
```

Comment:
Please name the file like the class it contains. So this file should probably be named "GLibSynchronizationContext".

Reply:
As the class integrates with the MainLoop how about calling it `MainLoopSynchronizationContext`?